extern/sector-storage/worker_local.go

```diff
@@ -607,17 +625,34 @@ func (l *LocalWorker) GenerateWindowPoSt(ctx context.Context, ppt abi.Registered
 	var slk sync.Mutex
 	var skipped []abi.SectorID
 
-	// todo throttle config
 	var wg sync.WaitGroup
 	wg.Add(len(sectors))
 
 	vproofs := make([][]byte, len(sectors))
 
 	for i, s := range sectors {
+		if l.challengeThrottle != nil {
```

Comment:
I think I'm missing something stupid but won't this block on the first sector?  The go routines below won't populate the channel for unblocking until they are spawned but the main loop needs to make it past this first to kick them off right?

Reply:
Oh, the channel ops are reversed